ui/app/components/app/modals/threebox-restore-confirm/threebox-restore-confirm.component.js

```diff
@@ -0,0 +1,58 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import Modal, { ModalContent } from '../../modal'
+
+export default class ThreeBoxRestoreConfirm extends PureComponent {
+  static propTypes = {
+    hideModal: PropTypes.func.isRequired,
+    restoreFromThreeBox: PropTypes.func.isRequired,
+    setThreeBoxSyncingPermission: PropTypes.func.isRequired,
+    setRestoredFromThreeBox: PropTypes.func.isRequired,
+    address: PropTypes.string.isRequired,
+  }
+
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  handleCancel = () => {
+    const { setRestoredFromThreeBox, hideModal } = this.props
+    setRestoredFromThreeBox(false)
+    hideModal()
+  }
+
+  handleConfirm = () => {
+    const {
+      restoreFromThreeBox,
+      setRestoredFromThreeBox,
+      setThreeBoxSyncingPermission,
+      hideModal,
+      address,
+    } = this.props
+    restoreFromThreeBox(address)
+      .then(() => {
+        setRestoredFromThreeBox(true)
```

Comment:
Does this need to be here at all? `restoreFromThreeBox` should already set this to true.

Reply:
Your right, `setRestoredFromThreeBox(true)` can be removed